src/cloud_provider/scaleway/kubernetes/mod.rs

```diff
@@ -513,6 +783,132 @@ impl<'a> Kapsule<'a> {
             return Err(e);
         }
 
+        let current_nodegroups = match self.get_existing_sanitized_node_groups() {
```

Comment:
Maybe worth switching to the new error methodo here. If you still want to return old EngineError, then just leverage `to_legacy_engine_error()`.
Here's an example: https://github.com/Qovery/engine/pull/583/files#diff-ac040a1944f39784cd97b35344b3698d4d4a703935a9f76d15ef00b8ebf9ba5fR338

Otherwise, I can do it it my PR after rebasing, just let me know.

Reply:
Logging PR as been merged. Please let me know if I can help. We can do it together if you want.